packages/vscode-extension/src/extension.ts

```diff
@@ -838,6 +775,44 @@ function registerDebugConfigProviders(context: vscode.ExtensionContext) {
   );
 }
 
+/**
+ * Tasks that doesn't block the user interaction so that they can be processed in background.
+ */
+async function runBackgroundAsyncTasks(
+  context: vscode.ExtensionContext,
+  isTeamsFxProject: boolean
+) {
+  ExtTelemetry.isFromSample = await handlers.getIsFromSample();
```

Comment:
If the variable is set after handlers.activate(), will the sent telemetry attach wrong properties?

Reply:
`isFromSample` is meaningful to teamsfx project and it will only happen to be undefined when user runs command like "build" before this code is executed. It should be very rare case. To fix it, we'd better make this sync function, and we can see whether it happens in telemetry later.

Reply:
There's also telemetry sent in handlers.activate for openTeamsApp

Reply:
Yes it's a bug and can be fixed in another PR, will check again after #5214 is merged.